packages/astro/src/core/endpoint/index.ts

```diff
@@ -83,6 +92,19 @@ export function createAPIContext({
 
 			return undefined;
 		},
+		get currentLocale(): string | undefined {
+			if (currentLocale) {
+				return currentLocale;
+			}
+			if (locales) {
+				currentLocale = computeCurrentLocale(request, locales, routingStrategy, defaultLocale);
+				if (currentLocale) {
+					return currentLocale;
+				}
+			}
+
+			return undefined;
```

Comment:
I believe this could be reduced to the following?

```suggestion
				return currentLocale;
			}
```